hooks/src/index.js

```diff
@@ -181,15 +181,43 @@ export function useReducer(reducer, initialState, init) {
 
 		hookState._component = currentComponent;
 
-		hookState._component.shouldComponentUpdate = () => {
-			if (!hookState._nextValue) return true;
+		if (!hookState._component._hasScuFromHooks) {
+			hookState._component.__hooks._hasScuFromHooks = true;
+			const prevScu = hookState._component.shouldComponentUpdate;
+			hookState._component.shouldComponentUpdate = (p, s, c) => {
+				const stateHooks = hookState._component.__hooks._list.filter(
+					x => x._component
+				);
+				const allHooksEmpty = stateHooks.every(x => !x._nextValue);
+				// When we have no updated hooks in the component we invoke the previous SCU or
+				// traverse the VDOM tree further.
+				if (allHooksEmpty) {
+					return prevScu ? prevScu(p, s, c) : true;
+				}
 
-			const currentValue = hookState._value[0];
-			hookState._value = hookState._nextValue;
-			hookState._nextValue = undefined;
+				// We check whether we have components with a nextValue set that
+				// have values that aren't equal to one another this pushes
+				// us to update further down the tree
+				const shouldSkipUpdating = stateHooks.every(hookItem => {
+					if (!hookItem._nextValue) return true;
```

Comment:
Per Marvin's comment above, shouldn't this check for `_nextValue` being undefined?

Reply:
> _nextValue in this case is an array like the normal state hook containing [value, updater] so using ! is the least bytes and correct as it's array or undefined